core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java

```diff
@@ -548,7 +548,7 @@ public Response getWebUIBrowse(@DefaultValue("/") @QueryParam("path") String req
         int offset = Integer.parseInt(requestOffset);
         int limit = Integer.parseInt(requestLimit);
         limit = offset == 0 && limit > fileInfos.size() ? fileInfos.size() : limit;
-        limit = offset * limit > fileInfos.size() ? fileInfos.size() % offset : limit;
+        limit = offset + limit > fileInfos.size() ? fileInfos.size() % offset : limit;
```

Comment:
I think we want `fileInfos.size() - offset` instead of `fileInfos.size() % offset`